### PR TITLE
Scoresheet-tabulation

### PR DESCRIPTION
**Pull request by bodiebice**
will provide further docs once ready to merge

### CODE REVIEW

File: emdcbackend/emdcbackend/views/scoresheets.py [hunk 1]
```diff
@@ -62,7 +76,21 @@ def update_scores(request):
     scores.field6 = request.data["field6"]
     scores.field7 = request.data["field7"]
     scores.field8 = request.data["field8"]
-    scores.field9 = request.data["field9"]
+    scores.fieldText = request.data["fieldText"]
+    if scores.sheet_type == ScoresheetEnum.PENALTIES:
+      scores.field9 = request.data["field9"]
```
Comment (KateGordon21) [inline emdcbackend/emdcbackend/views/scoresheets.py]: field 9 should be in the regular score sheet (comments field)

File: emdcbackend/emdcbackend/views/scoresheets.py [hunk 2]
```diff
@@ -45,6 +45,20 @@ def edit_score_sheet(request):
     scores.field6 = request.data["field6"]
     scores.field7 = request.data["field7"]
     scores.field8 = request.data["field8"]
+    if scores.sheet_type == ScoresheetEnum.PENALTIES:
+      scores.field9 = request.data["field9"]
```
Comment (KateGordon21) [inline emdcbackend/emdcbackend/views/scoresheets.py]: field 9 should be in regular score sheet

File: emdcbackend/emdcbackend/views/tabulation.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+from ..models import Teams, Scoresheet, MapScoresheetToTeamJudge, MapContestToTeam, ScoresheetEnum
+from ..serializers import TeamSerializer, ScoresheetSerializer
+
+# reference for this file's functions will be in a markdown file titled *Scoring Tabultaion Outline* in the onedrive
+
+@api_view(["PUT"])
+@authentication_classes([SessionAuthentication, TokenAuthentication])
+@permission_classes([IsAuthenticated])
+def tabulate_scores(request):
+  contest_team_ids = MapContestToTeam.objects.filter(request.data["id"])
```
Comment (KateGordon21) [inline emdcbackend/emdcbackend/views/tabulation.py]: is request.data[id] the contestid?

File: emdcbackend/emdcbackend/views/tabulation.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        scoresheets.append(tempscoresheet)
+      else:
+        return Response({"Error: Score Sheet Data Not Found from Mapping!"},status=status.HTTP_404_NOT_FOUND)
+    
+    # tabulation time!
+    totalscores = [0] * 12
+    for scoresheet in scoresheets:
+      if scoresheet.sheetType == ScoresheetEnum.PRESENTATION:
+        totalscores[0] = totalscores[0] + scoresheet.field1+ scoresheet.field2+ scoresheet.field3+ scoresheet.field4+ scoresheet.field5+ scoresheet.field6+ scoresheet.field7+ scoresheet.field8
+        totalscores[1] += 1
```
Comment (KateGordon21) [inline emdcbackend/emdcbackend/views/tabulation.py:52]: add comment that this is totaling up the # of score sheets

**Review by KateGordon21 — CHANGES_REQUESTED** [review]

File: emdcbackend/emdcbackend/views/tabulation.py (hunk 3 above)
Comment (bodiebice) [reply emdcbackend/emdcbackend/views/tabulation.py]: yeah, i'll rename it to be more clear

**Pull request merged**